restli-common/src/main/java/com/linkedin/restli/internal/common/URIMaskUtil.java

```diff
@@ -121,161 +121,161 @@ public static MaskTree decodeMaskUriFormat(StringBuilder toparse) throws Illegal
-        }
+          Integer maskValue;
+          if (toparse.charAt(position) == '-')
+          {
+            maskValue = MaskOperation.NEGATIVE_MASK_OP.getRepresentation();
+            position++;
+          }
+          else
+          {
+            maskValue = MaskOperation.POSITIVE_MASK_OP.getRepresentation();
+          }
 
-        int nextToken = -1;
-        StringBuilder field = new StringBuilder();
-        for (int ii = 0; ii < toparse.length(); ++ii)
-        {
-          char c = toparse.charAt(ii);
-          switch (c)
+          int nextToken = -1;
+          StringBuilder field = new StringBuilder();
+          for (int subPosition = position; subPosition < toparse.length(); ++subPosition)
           {
-          case ',':
-            state = ParseState.TRAVERSE;
-            nextToken = ii;
-            break;
-          case ':':
-            if (field.length() > 0 && (FilterConstants.START.equals(field.toString()) || FilterConstants.COUNT.equals(field.toString())))
+            char c = toparse.charAt(subPosition);
+            switch (c)
             {
-              if (!Character.isDigit(toparse.charAt(ii + 1)))
-              {
-                throw new IllegalMaskException("Malformed mask syntax: unexpected range value");
-              }
+              case ',':
+                state = ParseState.TRAVERSE;
+                nextToken = subPosition;
+                break;
+              case ':':
+                if (field.length() > 0 && (stringBuilderEquals(FilterConstants.START, field) || stringBuilderEquals(FilterConstants.COUNT, field)))
+                {
+                  if (!Character.isDigit(toparse.charAt(subPosition + 1)))
+                  {
+                    throw new IllegalMaskException("Malformed mask syntax: unexpected range value");
+                  }
 
-              ii++;
+                  subPosition++;
 
-              // Aggressively consume the numerical value for the range parameter as this is a special case.
-              StringBuilder rangeValue = new StringBuilder();
-              while (ii < toparse.length())
-              {
-                char ch = toparse.charAt(ii);
-                if (ch == ',')
-                {
-                  state = ParseState.TRAVERSE;
-                  nextToken = ii;
-                  break;
-                }
-                else if (ch == ')')
-                {
-                  state = ParseState.ASCEND;
-                  nextToken = ii;
-                  break;
-                }
-                else if (Character.isDigit(ch))
-                {
-                  rangeValue.append(ch);
+                  // Aggressively consume the numerical value for the range parameter as this is a special case.
+                  maskValue = 0;
+                  while (subPosition < toparse.length())
+                  {
+                    char ch = toparse.charAt(subPosition);
+                    if (ch == ',')
```

Comment:
Switch statements are faster than if-else.

Reply:
Can't use a switch statement because one of the conditions is `Character.isDigit(ch)`. Additionally, if else statements and switch statements are compiled identically until the number of conditions hits 5, and then switch statements get compiled to a hash table